packages/frontend/app/(dashboard)/components/ImageCropperModal.tsx

```diff
@@ -0,0 +1,166 @@
+  const [imageSrc, setImageSrc] = useState<string | null>(null)
+
+  const onCropCompleteCallback = useCallback(
+    (croppedArea: CroppedArea, croppedAreaPixels: CroppedAreaPixels) => {
+      setCroppedAreaPixels(croppedAreaPixels)
+    },
+    [],
+  )
+
+  // MIME types that are compatible as inputs with the sharp library in the backend
+  const viableFileTypes = [
+    'image/jpeg',
+    'image/png',
+    'image/webp',
+    'image/avif',
+    'image/gif',
+    'image/svg+xml',
+    'image/tiff',
+  ]
+
+  const beforeUpload = (file: any): boolean => {
+    const isValidMimeType = viableFileTypes.includes(file.type)
+
+    if (!isValidMimeType) {
+      message.error('You can only upload JPGs or PNGs!')
+    }
+
+    const isLt5M = file.size / 1024 / 1024 < 5
+    if (!isLt5M) {
+      message.error('Image must smaller than 5MB!')
+    }
+
+    return isValidMimeType && isLt5M
+  }
+
+  const handleUpload = async (file: any): Promise<void> => {
+    const reader = new FileReader()
+    reader.onload = () => {
+      setImageSrc(reader.result as string)
+      setImageType(file.type)
+      setFileName(file.name)
+    }
+    reader.readAsDataURL(file)
+  }
+
+  const handleCropComplete = async () => {
+    if (!imageSrc || !croppedAreaPixels || !imageType) {
+      return
+    }
+    try {
+      setUploading(true) // Start the upload state
+      const croppedImage = await getCroppedImg(
+        imageSrc!,
+        croppedAreaPixels!,
+        imageType!,
+      )
+      const formData = new FormData()
+      formData.append('file', croppedImage)
+      const response = await fetch(postURL, {
+        method: 'POST',
+        body: formData,
+      })
+      const data = await response.json()
+      if (response.ok) {
+        message.success(`${fileName} file uploaded successfully`)
+        onUpdateComplete(data.fileName)
+      } else {
+        message.error(`${fileName} file upload failed: ${data.message}`)
+      }
+    } catch (error) {
+      message.error(`Error uploading ${fileName}. Please try again.`)
+    } finally {
+      setUploading(false) // Reset the upload state regardless of success or error
+      onCancel() // Close the modal
+    }
+  }
+
+  return (
+    <Modal
+      title={`Crop ${imgName}`}
+      open={isOpen}
+      onOk={handleCropComplete}
+      onCancel={() => onCancel()}
+      destroyOnClose={true}
+      width="auto"
+      className="flex flex-col items-center justify-center"
+    >
+      <Upload
+        customRequest={async ({ file }) => await handleUpload(file)}
+        beforeUpload={beforeUpload}
+        className="mb-2"
+        showUploadList={false}
+        maxCount={1}
+      >
+        <button className="min-w-[85vw] rounded-lg border-2 bg-white p-2 lg:min-w-[30vw]">
+          <UploadOutlined />
+          <span className="ml-2">{`Upload ${imgName}`}</span>
+        </button>
+      </Upload>
+      <div className="relative mt-2 h-[300px] w-full">
```

Comment:
```suggestion
      <div className="relative mt-2 h-80 w-full">
```
I suggest using relative units where possible. Though if this is a component should always be 300px than you can disregard this comment.